app/src/main/java/com/example/diceroller/RolledDiceViewModel.kt

```diff
@@ -1,65 +0,0 @@
-package com.example.diceroller
-
-import androidx.lifecycle.ViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import java.text.DateFormat
-import kotlin.random.Random
-
-class RolledDiceViewModel : ViewModel() {
-    private val _currentDice: MutableStateFlow<DiceType> = MutableStateFlow(DiceType.D6)
-    val currentDice = _currentDice.asStateFlow()
-
-    private val _listOfDices: MutableStateFlow<List<Dice>> = MutableStateFlow(listOf<Dice>())
-    val listOfDices = _listOfDices.asStateFlow()
-
-    fun getHistoryOfDices(): List<Dice> = listOfDices.value
-
-    fun addDiceToList(dice: Dice) {
-        _listOfDices.value = listOfDices.value+dice
-    }
-
-    //    Deprecated, changed to triggers
-    fun switchCurrentDiceType(diceType: DiceType) {
-        _currentDice.value = diceType
-    }
-
-    fun getCurrentDiceType(): DiceType = currentDice.value
-
-
-
-    fun rollDice():Dice {
```

Comment:
🙅  I can probably speak more on this but I'll try to keep it simple. Remember that Views are dumb. Your Fragment should be so dumb it doesn't know how to flip a light switch without being told how to. 

You also can't have your Fragment telling its ViewModel what to do. Your ViewModel has all the knowledge and by extension all the power.

With `rollDice()` the Fragment is telling the ViewModel what to do. It is saying "Hey ViewModel! Roll the dice!" to make it dumb you want it to say "Derr uh hey ViewModel, the user clicked on the roll button". Then the ViewModel being all wise will do its thing, it will grab a die, roll it, and spit out a new State. Emitting a State is the ViewModel saying "Here are some easy instructions for you to follow, View". The instructions come in a way that the Fragment is so dumb, it doesn't know why it got these instructions, it just follows them, instructions always come from the same place, it says do the thing, i do the thing.

I hope that the refactors I made demonstrate this.